src/prefect/futures.py

```diff
@@ -104,18 +106,23 @@ class PrefectFuture(Generic[R, A]):
 
     def __init__(
         self,
-        task_run: TaskRun,
-        run_key: str,
+        name: str,
+        key: UUID,
```

Comment:
I changed key to be `f"{flow_run_context.flow_run.id}-{task.task_key}-{dynamic_key}"
    )` in line 880 of `engine.py` and forgot to change the types here.

Are we alright with that? I think it makes sense given it's a unique key for the task.
But hmmm perhaps we need to include flow and task retry counts too?

Reply:
Task retries will occur _within_ the submitted call. Flow run retries would need to be included though. I don't think we should mix real information about the task into this key. This should be a unique identifier to facilitate lookup of the future's result and as you've noted collisions are a concern.

Reply:
Hmm okay should we change this back to a `uuid` then?